### PR TITLE
Graph RBAC: add thumbprint in certificate definition

### PR DESCRIPTION
**Pull request by ChenTanyi**
fix https://github.com/Azure/azure-libraries-for-net/issues/915

### CODE REVIEW

File: src/ResourceManagement/Graph.RBAC/Domain/ICertificateCredential.cs [hunk 1]
```diff
@@ -13,5 +13,11 @@ public interface ICertificateCredential :
         Microsoft.Azure.Management.Graph.RBAC.Fluent.ICredential,
         Microsoft.Azure.Management.ResourceManager.Fluent.Core.IHasInner<Models.KeyCredential>
     {
+        /// <summary>
+        /// Custom Key Identifier. If the credential is defined by .NET SDK with name,
+        /// it would be the base64 encoding of name. If it is set by other tools, it would
+        /// be that value. Otherwise, it would usually be the thrumbprint of cretificate.
```
Comment (weidongxu-microsoft) [inline src/ResourceManagement/Graph.RBAC/Domain/ICertificateCredential.cs]: spelling: thrumbprint and cretificate

Comment (ChenTanyi) [reply src/ResourceManagement/Graph.RBAC/Domain/ICertificateCredential.cs]: done

**Comment by ChenTanyi** [comment]
@weidongxu-microsoft @yaohaizh I add the deleting of these credential, you could review again.

File: src/ResourceManagement/Graph.RBAC/CertificateCredentialImpl.cs [hunk 2]
```diff
@@ -140,11 +140,12 @@ internal CertificateCredentialImpl(KeyCredential keyCredential)
             }
         }
 
-        internal CertificateCredentialImpl(string name, IHasCredential<T> parent, string customKeyIdentifier)
+        internal CertificateCredentialImpl(string name, IHasCredential<T> parent, string customKeyIdentifier, Guid? keyId)
             : base(name, new KeyCredential()
             {
                 Usage = "Verify",
                 CustomKeyIdentifier = customKeyIdentifier,
+                KeyId = (keyId ?? Guid.NewGuid()).ToString(),
```
Comment (weidongxu-microsoft) [inline src/ResourceManagement/Graph.RBAC/CertificateCredentialImpl.cs]: So previous KeyId is null. And service would generate something for it upon PUT?

Comment (ChenTanyi) [reply src/ResourceManagement/Graph.RBAC/CertificateCredentialImpl.cs]: Yeah, the service will generate a UUID as KeyID, if it is null. The azure-cli generate keyId by itself, so I think this behavior would not be changed. https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/role/custom.py#L1604

**Pull request merged**